composeApp/src/iosMain/kotlin/org/ooni/probe/background/RunOperation.kt

```diff
@@ -0,0 +1,59 @@
+package org.ooni.probe.background
+
+import co.touchlab.kermit.Logger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.takeWhile
+import kotlinx.coroutines.launch
+import org.ooni.probe.data.models.RunSpecification
+import org.ooni.probe.data.models.TestRunState
+import org.ooni.probe.domain.GetAutoRunSpecification
+import org.ooni.probe.domain.RunDescriptors
+import platform.Foundation.NSOperation
+
+class RunOperation(
+    private val spec: RunSpecification? = null,
+    private val getAutoRunSpecification: GetAutoRunSpecification? = null,
+    private val runDescriptors: RunDescriptors,
+    private val getCurrentTestState: () -> Flow<TestRunState>,
+) : NSOperation() {
+    init {
+        require(spec != null || getAutoRunSpecification != null) {
+            "Either spec or getAutoRunSpecification must be provided"
+        }
+    }
+
+    override fun main() {
+        Logger.d { "Running operation" }
+        val coroutineScopeScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
```

Comment:
Should this be running on the main thread? I know we switch to the background dispatcher on the engine and DB, but just wondering if we need to specify Main here, or we can just go with the Default.

Reply:
Updated